packages/wrangler/src/__tests__/docs.test.ts

```diff
@@ -0,0 +1,122 @@
+	const std = mockConsoleMethods();
+
+	beforeEach(() => {
+		globalThis.ALGOLIA_APP_ID = "FAKE-ID";
+		globalThis.ALGOLIA_PUBLIC_KEY = "FAKE-KEY";
+
+		msw.use(
+			http.post<Record<string, never>, { params: string | undefined }>(
+				`*/1/indexes/developers-cloudflare2/query`,
+				async ({ request }) => {
+					const json = await request.json();
+					const params = new URLSearchParams(json.params);
+					return HttpResponse.json({
+						hits: [
+							{
+								url: "FAKE_DOCS_URL?" + params.get("query"),
+							},
+						],
+					});
+				},
+				{ once: true }
+			)
+		);
+	});
+
+	afterEach(() => {
+		delete globalThis.ALGOLIA_APP_ID;
+		delete globalThis.ALGOLIA_PUBLIC_KEY;
+	});
+
+	test("--help", async ({ expect }) => {
+		const result = runWrangler("docs --help");
+
+		await expect(result).resolves.toBeUndefined();
+		expect(std.out).toMatchInlineSnapshot(`
+			"wrangler docs [command..]
+
+			📚 Open Wrangler's command documentation in your browser
+
+
+			POSITIONALS
+			  command  Enter the wrangler command you want to know more about  [array] [default: []]
+
+			GLOBAL FLAGS
+			  -j, --experimental-json-config  Experimental: support wrangler.json  [boolean]
+			  -c, --config                    Path to .toml configuration file  [string]
+			  -e, --env                       Environment to use for operations and .env files  [string]
+			  -h, --help                      Show help  [boolean]
+			  -v, --version                   Show version number  [boolean]
+
+			OPTIONS
+			  -y, --yes  Takes you to the docs, even if search fails  [boolean]"
+		`);
+	});
+
+	test("opens a browser to Cloudflare docs when given no search term", async ({
+		expect,
+	}) => {
+		await runWrangler("docs");
+		expect(std).toMatchInlineSnapshot(`
+			Object {
+			  "debug": "",
+			  "err": "",
+			  "info": "",
+			  "out": "Opening a link in your default browser: https://developers.cloudflare.com/workers/wrangler/commands/",
+			  "warn": "",
+			}
+		`);
+		expect(openInBrowser).toHaveBeenCalledWith(
+			"https://developers.cloudflare.com/workers/wrangler/commands/"
+		);
+	});
+
+	test("opens a browser to Cloudflare docs when given a single search term", async ({
+		expect,
+	}) => {
+		await runWrangler("docs dev");
+		expect(std).toMatchInlineSnapshot(`
+			Object {
+			  "debug": "",
+			  "err": "",
+			  "info": "",
+			  "out": "Opening a link in your default browser: FAKE_DOCS_URL?dev",
+			  "warn": "",
+			}
+		`);
+		expect(openInBrowser).toHaveBeenCalledWith("FAKE_DOCS_URL?dev");
+	});
+
+	test("opens a browser to Cloudflare docs when given multiple search terms", async ({
+		expect,
+	}) => {
+		await runWrangler("docs foo bar");
+		expect(std).toMatchInlineSnapshot(`
+			Object {
+			  "debug": "",
+			  "err": "",
+			  "info": "",
+			  "out": "Opening a link in your default browser: FAKE_DOCS_URL?foo bar",
+			  "warn": "",
+			}
+		`);
+		expect(openInBrowser).toHaveBeenCalledWith("FAKE_DOCS_URL?foo bar");
```

Comment:
Is `?foo bar` valid?

Reply:
This string is not actually what is passed to `openInBrowser()` the actual URL will be the first result returned from the Algolia search. This string is just the fake response I cobbled together in the MSW fetch handler so that we could see what is being passed to Algolia.